book/src/snapshot-verification.md

```diff
@@ -0,0 +1,47 @@
+# Verifying Snapshots and Account state proof
+
+## Problem
+
+When a validator boots up from a snapshot, it needs a way to verify the account set matches what the rest of the network sees quickly. A potential attacker
+could give the validator an incorrect state, and then try to convince it to accept a transaction that would otherwise be rejected.
+
+## Solution
+
+Currently the bank hash is derived from hashing the delta state of the accounts in a slot which is then combined with the previous bank hash value. The problem with this is that the list of hashes
+will grow on the order of the number of slots processed by the chain and become a burden to both transmit and verify successfully.
+
+Another naive method could be to create a merkle tree of the account state. This has the downside that with each account update which removes an account state from the tree, the merkle
+tree would have to be recomputed from the entire account state of all live accounts in the system.
+
+To verify the snapshot, we propose the following:
+
+On account store hash the following data:
+* Account owner
+* Account data
+* Account pubkey
+* Account lamports balance
+* Fork the account is stored on
+
+Use this resulting hash value as input to an expansion function which expands the hash value into an image value.
+The function will create a 440 byte block of data where the first 32 bytes are the hash value, and the next 440 - 32 bytes
+are generated from a Chacha RNG with the hash as the seed.
+
+The account images are then combined with xor. The previous account value will be xored into the state and the new
+account value also xored into the state.
+
+Voting and sysvar hash values occur with the hash of the resulting full image value.
```

Comment:
Do we also want to show some threshold of the network has voted on/confirmed the resulting image value?

Reply:
That would be shown when the validator downloads the snapshot?

Reply:
I think that makes sense. So a snapshot will be augmented with the expected results of this new XOR calculation, which a validator will verify after it's unpacked the bank in the snapshot.

But the threshold proof will be provided by another validator as the snapshot is being downloaded. Just have to make sure we check the threshold proof is for the same XOR image as the one packaged in the snapshot.

Reply:
What does the threshold proof look like?

Reply:
I'm think we can just reuse @t-nelson's design for SPV we should have votes showing confirmations on a bank.